### ISSUE TITLE
GDScript 2.0: Accessing a property of built-in typed variable inside its setter function returns `null`

### ISSUE THREAD

**Issue by xorborg** [godotengine/godot#51175]
### Godot version

4.0.dev.custom_build[854725f40]

### System information

Arch Linux

### Issue description

Accessing the properties of a built-in type inside a setter function returns null.

Calling constructors inside the setter results in strange behavior.

Without using static typing it works as expected.

### Steps to reproduce

The project below contains a single node with following script:

```
extends Node
var vector := Vector2(1,1):
  set(new_vector):
  #accessing x returns null
    print("vector: ",vector)
    print("vector.x: ",vector.x)
    print("new_vector: ",new_vector)
    print("new_vector.x: ",new_vector.x)

  #calling constructors leads to weird behavior
    Vector2i(new_vector)
    print("\nvector.x: ",vector.x)
    print("new_vector.x: ",new_vector.x)
    Rect2(0,0,0,0)
    print("vector.x: ",vector.x)
    print("new_vector.x: ",new_vector.x)
    Vector2(new_vector).x
    print("vector.x: ",vector.x)
    print("new_vector.x: ",new_vector.x)

func _ready():
  self.vector = Vector2(2,2)
```

Running it prints:

```
vector: (1, 1)
vector.x: Null
new_vector: (2, 2)
new_vector.x: Null

vector.x: (0, 1072693248)
new_vector.x: Null
vector.x: [P: (0, 1.875), S: (0, 0)]
new_vector.x: (0, 1073741824)
vector.x: 1
new_vector.x: (0, 2)
```

### Minimal reproduction project

[setterpropertytest.zip](https://github.com/godotengine/godot/files/6916681/setterpropertytest.zip)

**Comment by cdemirer** [comment]
This might have been fixed, can't reproduce on master.

**Comment by akien-mga** [comment]
Indeed, I could confirm the bug in 4.0-dev.20210727, and it's fixed in current `master` (35fff9a55e61934c84bc402a5a94073496d60709).
